rasterio/_filepath.pyx

```diff
@@ -138,40 +135,33 @@ cdef void* filepath_open(void *pUserData, const char *pszFilename, const char *p
     cdef dict filesystem_info = <object>pUserData
 
     try:
-        file_wrapper = clone_file_wrapper(filesystem_info[pszFilename])
+        file_obj = clone_file_obj(filesystem_info[pszFilename])
     except KeyError:
         log.info("Object not found in virtual filesystem: filename=%r", pszFilename)
         return NULL
 
-    if not hasattr(file_wrapper, "_file_obj"):
-        log.error("Unexpected file object found in FilePath filesystem.")
-        return NULL
-
     # Open file wrappers are kept in this set and removed when closed.
-    _OPEN_FILE_WRAPPERS.add(file_wrapper)
+    _OPEN_FILE_OBJS.add(file_obj)
 
-    return <void *>file_wrapper
+    return <void *>file_obj
 
 ## File functions
 
 cdef vsi_l_offset filepath_tell(void *pFile) with gil:
-    cdef object file_wrapper = <object>pFile
-    cdef object file_obj = file_wrapper._file_obj
+    cdef object file_obj = <object>pFile
```

Comment:
The thing we're giving up is that we no longer know the fileobj's VSI filename within the callback. I'm not sure how useful that is. We didn't use it before. And if we did need it, we could come up with a lighter wrapper than the FilePath class.

Reply:
I think as long as the high level stuff knows and GDAL knows then maybe it doesn't matter. If I'm remembering this well enough, I think this really only takes away from flexibility in the future with throwing things on the python wrapper class rather than only having access to the object the user provided.